### PR TITLE
add profanity check for sublets

### PR DESCRIPTION
**Pull request by ashleyzhang01**

### CODE REVIEW

File: backend/sublet/serializers.py [hunk 1]
```diff
@@ -91,6 +92,21 @@ class Meta:
...
+        return value
+
+    def validate_description(self, value):
+        # Check for profanity in the description
+        if self.contains_profanity(value):
+            raise serializers.ValidationError("The description contains inappropriate language.")
+        return value
+
+    def contains_profanity(self, text):
+        return predict([text])[0] == 1
```
Comment (vcai122) [inline backend/sublet/serializers.py:108]: `nit`: don't need == 1 since 1==True and 0==False

File: backend/Pipfile [hunk 2]
```diff
@@ -46,6 +46,8 @@ python-dateutil = "*"
 selenium = "*"
 webdriver-manager = "*"
 pre-commit = "*"
+profanity-check = "*"
+alt-profanity-check = "*"
```
Comment (vcai122) [inline backend/Pipfile:50]: why do we need both? on alt-profanity-check's website it says: "Alt profanity check is a drop-in replacement of the profanity-check library for the not so well maintained", so it seems like just using alt-profanity-check should be fine.

File: backend/sublet/serializers.py [hunk 3]
```diff
@@ -91,6 +92,21 @@ class Meta:
...
         ]
 
+    def validate_title(self, value):
+        # Check for profanity in the title
+        if self.contains_profanity(value):
+            raise serializers.ValidationError("The title contains inappropriate language.")
+        return value
+
+    def validate_description(self, value):
+        # Check for profanity in the description
```
Comment (vcai122) [inline backend/sublet/serializers.py:102]: lmao huge nit but i would argue doesn't need comments since its already readable

**Review by vcai122 — CHANGES_REQUESTED** [review]
mainly just the pipfile comment, but looks pretty good otherwise! 0 shots

File: backend/Pipfile (hunk 2 above)
Comment (ashleyzhang01) [reply backend/Pipfile:50]: i had some weird installation errors without the alt-profanity-check, and it wouldn't run/pipfile wouldn't lock. online fixes said to do this

Comment (ashleyzhang01) [reply backend/Pipfile:50]: oh wait jk getting rid of the first ill do that

File: backend/sublet/serializers.py [hunk 4]
```diff
@@ -91,6 +92,19 @@ class Meta:
...
+            raise serializers.ValidationError("The title contains inappropriate language.")
+        return value
+
+    def validate_description(self, value):
+        if self.contains_profanity(value):
+            raise serializers.ValidationError("The description contains inappropriate language.")
+        return value
+
+    def contains_profanity(self, text):
+        return predict([text])[0]
```
Comment (judtinzhang) [inline backend/sublet/serializers.py:106]: to confirm, do we need to lowercase the text or anything? or is this package already handle this

**Review by judtinzhang — APPROVED** [review]
LGTM! One quick question. Thanks ashley, 0 shots

File: backend/sublet/serializers.py (hunk 4 above)
Comment (ashleyzhang01) [reply backend/sublet/serializers.py:106]: it's handled

**Review by vcai122 — APPROVED** [review]
LGTM!!!!!!!!!!!!!!!!

**Pull request merged**